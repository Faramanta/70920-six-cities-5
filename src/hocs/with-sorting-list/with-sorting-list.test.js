import renderer from 'react-test-renderer';
import {Fragment} from 'react';
import withOpenSortingList from "./with-sorting-list";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withOpenSortingList(MockComponent);

it(`withOpenSortingList is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      isOpen={false}
      onOpenList={() => {}}
    >
      <Fragment />
    </MockComponentWrapped>
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
