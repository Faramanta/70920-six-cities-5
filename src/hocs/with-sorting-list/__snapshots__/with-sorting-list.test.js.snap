// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withOpenSortingList is rendered correctly 1`] = `<div />`;
